### PR TITLE
[Merged by Bors] - refactor(UniformSpace): drop `separationRel`

### LINKED ISSUE #2031: refactor: drop `UniformSpace.separationRel`
Use `Inseparable` instead. These relations are propositionally equal.

### PR DESCRIPTION
**Pull request by urkud**
We had duplicated API between topological spaces and uniform spaces.
In this PR I mostly deduplicate it with some exceptions:

- `SeparationQuotient.lift'` and `SeparationQuotient.map`
  are leftovers from the old version
  that are designed to work with uniform spaces;
- probably, some theorems/instances still assume `UniformSpace`
  when `TopologicalSpace` would work.

Outside of `UniformSpace/Separation`, I mostly changed `SeparatedSpace` to `T0Space`
and `separationRel` to `Inseparable`. I also rewrote a few proofs that were broken by the API change.

Fixes #2031

---

- [x] depends on: #10871

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by ADedecker — COMMENTED** [review]
Thanks a lot for all this work! I think I've looked at everything but it would be nice to get another look by someone else, maybe @PatrickMassot ?

File: Mathlib/Topology/Algebra/Group/Basic.lean [hunk 1]
```diff
@@ -1580,6 +1580,14 @@ instance (priority := 100) TopologicalGroup.regularSpace : RegularSpace G := by
 
 variable {G}
 
+@[to_additive]
+theorem group_inseparable_iff {x y : G} : Inseparable x y ↔ x / y ∈ closure (1 : Set G) := by
+  rw [← singleton_one, ← specializes_iff_mem_closure, specializes_comm, specializes_iff_inseparable,
+    ← (Homeomorph.mulRight y⁻¹).embedding.inseparable_iff]
+  simp [div_eq_mul_inv]
+#align group_separation_rel group_inseparable_iff
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Topology/Algebra/Group/Basic.lean:1588]: IMHO, moving to a file that is imported by the original file should be OK.

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 2]
```diff
@@ -320,6 +323,16 @@ theorem UniformInducing.isComplete_range [CompleteSpace α] {f : α → β} (hf
...
 
+theorem SeparationQuotient.completeSpace_iff :
+    CompleteSpace (SeparationQuotient α) ↔ CompleteSpace α := by
+  rw [completeSpace_iff_isComplete_univ, ← range_mk,
+    ← completeSpace_iff_isComplete_range uniformInducing_mk]
+
+instance SeparationQuotient.instCompleteSpace [CompleteSpace α] :
+    CompleteSpace (SeparationQuotient α) :=
+  completeSpace_iff.2 ‹_›
+#align uniform_space.complete_space_separation SeparationQuotient.instCompleteSpace
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Topology/UniformSpace/UniformEmbedding.lean:334]: Same argument as above: anyone who imports the old location gets the new instance.

File: Mathlib/Topology/Algebra/Group/Basic.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (ADedecker) [reply Mathlib/Topology/Algebra/Group/Basic.lean:1588]: I was thinking of a potential issue with mathport, but I've looked at older discussions and it should be fine (I could have guessed actually, we've been moving aligns regularly...)

File: Mathlib/Topology/Algebra/UniformGroup.lean [hunk 3]
```diff
@@ -374,15 +374,6 @@ theorem Filter.HasBasis.uniformity_of_nhds_one_inv_mul_swapped {ι} {p : ι →
 #align filter.has_basis.uniformity_of_nhds_one_inv_mul_swapped Filter.HasBasis.uniformity_of_nhds_one_inv_mul_swapped
 #align filter.has_basis.uniformity_of_nhds_zero_neg_add_swapped Filter.HasBasis.uniformity_of_nhds_zero_neg_add_swapped
 
-@[to_additive]
-theorem group_separationRel (x y : α) : (x, y) ∈ separationRel α ↔ x / y ∈ closure ({1} : Set α) :=
```
Comment (PatrickMassot) [inline Mathlib/Topology/Algebra/UniformGroup.lean:378]: Aren’t we loosing some mathematical content here?

File: Mathlib/Topology/Algebra/UniformRing.lean [hunk 4]
```diff
@@ -236,38 +237,51 @@ namespace UniformSpace
 
 variable {α : Type*}
 
-theorem ring_sep_rel (α) [CommRing α] [UniformSpace α] [UniformAddGroup α] [TopologicalRing α] :
-    separationSetoid α = Submodule.quotientRel (Ideal.closure ⊥) :=
+-- TODO: move (some of) these results to the file about topological rings
+theorem ring_sep_rel (α) [CommRing α] [TopologicalSpace α] [TopologicalRing α] :
```
Comment (PatrickMassot) [inline Mathlib/Topology/Algebra/UniformRing.lean:241]: Shouldn’t we rename that theorem?

File: Mathlib/Topology/UniformSpace/Separation.lean [hunk 5]
```diff
@@ -81,143 +81,86 @@ variable [UniformSpace α] [UniformSpace β] [UniformSpace γ]
...
+  .ofBasis
+    (fun _ => nhds_basis_uniformity' uniformity_hasBasis_closed)
+    fun _ _ h => h.2.preimage <| continuous_const.prod_mk continuous_id
 #align uniform_space.to_regular_space UniformSpace.to_regularSpace
 
--- porting note: todo: use `Inseparable`
-/-- The separation relation is the intersection of all entourages.
-  Two points which are related by the separation relation are "indistinguishable"
-  according to the uniform structure. -/
-def separationRel (α : Type u) [UniformSpace α] := (𝓤 α).ker
```
[the remark this answers is not in the document]
Comment (PatrickMassot) [reply Mathlib/Topology/UniformSpace/Separation.lean:93]: Really the whole module docstring needs to be updated. And I’m not sure where we currently find that for uniform space being T0 is equivalent to being T2.

File: Mathlib/Topology/Algebra/UniformGroup.lean (hunk 3 above)
Comment (ADedecker) [reply Mathlib/Topology/Algebra/UniformGroup.lean:378]: I think the new `group_inseparable_iff` takes care of that.

Comment (PatrickMassot) [reply Mathlib/Topology/Algebra/UniformGroup.lean:378]: Ok great. I looked for it but couldn’t find it.

File: Mathlib/Topology/UniformSpace/Separation.lean (hunk 5 above)
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Topology/UniformSpace/Separation.lean:93]: I'll rewrite the module docstring tonight or tomorrow. Typeclass inference now can prove `[T0Space X] [R1Space X] : T2Space X`. Do you want me to add it as an `Iff` lemma?

[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Topology/UniformSpace/Separation.lean:93]: Done.

File: Mathlib/Topology/Algebra/UniformRing.lean (hunk 4 above)
Comment (urkud) [reply Mathlib/Topology/Algebra/UniformRing.lean]: Done.

**Comment by ADedecker** [comment]
Thanks a lot!

bors merge

**Pull request closed without merging**